### DISCUSSION TITLE
Localization of a Game score

### DISCUSSION THREAD

**Discussion by PicoAnimate** [MonoGame/MonoGame#8541] [General]
I am writing a Monogame game. The users score usually ranges from 1 to 50,000,000 (One to Fifty Million As in the USA format).
It recently came to my attention that numbers are formatted in different formats depending upon the culture. 
For example in the USA English system the number 123,456.79 would be written as
                                                                                  123.456,79 (In Germany)
                                                                                  123 456,79 (In French)
Luckily C# has some built in formatting using System.Globalization.CultureInfo.GetCultureInfo if I desire to format the score for other cultures.

So here is my question, are Game scores typically formatted for display for other cultures? I have no idea, since I have never downloaded a game in a different country to see if the score is formatted to that countries culture format.

I want my game to be enjoyed by the users of different countries, so that they have a great gaming experience. So should I format the score according to a specific culture? Maybe I can hear back from some experienced game developers in this area, or even people from different cultures. Thank you for your help.

**Comment by ThomasFOG** [comment]
It depends.

Typically, if a number is the transcript of what a character is saying, you might want to localize the formatting according to the display language. But not always... for accessibility concerns, some numbers are said in text (e.g. "one million" instead of "1000000"), because of dyslexia and other things... but other numbers are in plain without formatting because they are easier to read and memorize (e.g. "bring me 32 flowers").

When it's not a transcript (e.g. UI), it's case by case that also depends on the art direction. The one thing we usually localize is the decimal separator but thousands separators aren't always (decimal numbers are usually entirely avoided for accessibility reasons). Here you usually want to maximise accessibility of information within your art direction over linguistic correctness. It also depends on how large the numbers are. It might also depends on what's diegitically displaying that number (e.g. do you consider the score to be displayed by a computer manufactured in the US in your game's lore no matter the display language? it would make sense to use the US formatting there). Sometime you need numbers to be big, well detached and readable (e.g. player HP), and sometime when you need large numbers just to enhance the game feel (e.g. very large damage numbers flying around) formatting isn't important and can even be funny/unreadable. There's no golden rule.

**Reply by PicoAnimate** [reply]
Thanks for the reply, however I still feel like I don't have a clear answer on this, so maybe you or another member can help me to understand based upon a hypothetical example. (Bear with me please, let's simplify this with an example).

Suppose you have a poker game where each players total value in poker chips is displayed in a phone app. The chip total for each player is displayed above an image of the player. In this example there are 4 players.

Player 1 only has 300 chips remaining, Player #2 has 1,000,000 chips Player #3 has 3,500,000 chips, and player #4 has 8,780,000 chips.

Observation :
As a phone app user from the USA, these numbers are easy to read, because we are taught to separate the digit groupings using a comma. However in (Dutch, German, French for example) a period would be used instead of a comma.

So to accommodate and to be respectful of other cultures, should I display the numbers according to that cultures number formatting?
My end goal is user retention, and usability. If they find USA number style formatting confusing they might very well delete the app, or just choose to never play it again.

**Comment by MonogameGuy** [comment]
There is no reason you cannot use the culture information to format the numbers.  

Be careful when it comes to currency, as all of the symbols must be included in the `SpriteFont` and no value conversions will take place.

For in-game fake money, I would hard-code the currency sign as dollar or euro.  This prevents non-nonsensical amounts like 1 yen.
